src/__swaps__/screens/Swap/hooks/useSearchCurrencyLists.ts

```diff
@@ -23,156 +22,161 @@ export interface AssetToBuySection {
   id: AssetToBuySectionId;
 }
 
-const filterBridgeAsset = ({ asset, filter = '' }: { asset?: SearchAsset; filter?: string }) =>
+const filterBridgeAsset = ({ asset, filter = '' }: { asset: SearchAsset | null | undefined; filter: string }) =>
   asset?.address?.toLowerCase()?.startsWith(filter?.toLowerCase()) ||
   asset?.name?.toLowerCase()?.startsWith(filter?.toLowerCase()) ||
   asset?.symbol?.toLowerCase()?.startsWith(filter?.toLowerCase());
 
 export function useSearchCurrencyLists() {
   const { internalSelectedInputAsset: assetToSell, selectedOutputChainId } = useSwapContext();
 
-  const searchQuery = userAssetsStore(state => state.searchQuery);
+  const query = useSwapsStore(state => state.outputSearchQuery.trim().toLowerCase());
 
-  const [inputChainId, setInputChainId] = useState(assetToSell.value?.chainId ?? ChainId.mainnet);
-  const [toChainId, setToChainId] = useState(selectedOutputChainId.value);
-  const [assetToSellAddress, setAssetToSellAddress] = useState(
-    assetToSell.value?.[assetToSell.value?.chainId === ChainId.mainnet ? 'address' : 'mainnetAddress']
-  );
+  const [state, setState] = useState({
+    assetToSellAddress: assetToSell.value?.[assetToSell.value?.chainId === ChainId.mainnet ? 'address' : 'mainnetAddress'],
+    fromChainId: assetToSell.value?.chainId ?? undefined,
+    isCrosschainSearch: assetToSell.value ? assetToSell.value.chainId !== selectedOutputChainId.value : false,
+    toChainId: selectedOutputChainId.value ?? ChainId.mainnet,
+  });
 
-  const query = useMemo(() => searchQuery.trim().toLowerCase(), [searchQuery]);
-  const enableUnverifiedSearch = useMemo(() => searchQuery.length > 2, [searchQuery]);
+  const debouncedStateSet = useDebouncedCallback(setState, 500, { leading: true, trailing: true });
```

Comment:
see comment above, but this is going to be problematic since we're not keeping the previous reference to the old state.